### DISCUSSION TITLE
Update a named field in a tuple property

### DISCUSSION THREAD

**Discussion by realTPowell** [geldata/gel#5397] [Q&A] [answered]
I'm having trouble working out how to elegantly update single fields in a tuple property. As an example, take the following property:
```
type Character { 
  property stats -> tuple<Strength: int16, Dexterity: int16, Wisdom:  @int16>
}
``` 
I want to update one field of this stat-set, but it seems that although the 'dot' syntax is used to access tuple fields, it does not combine with path resolution, which can only terminate in a link or property. The seemingly intuitive query:
```
update Character
set {stats.Strength := 1}
```
raises the error `EdgeQLSyntaxError: Unexpected '.'`, and 

```
update Character.stats
set {Strength := 1}
```
raises `QueryError: cannot update non-ObjectType objects`. It therefore seems I have to update the whole tuple set at once, but to only affect one field this will require some gymnastics with 'with' clauses that seems too inelegant to be the best option.

I can see a couple of potential solutions:

1. Model each field of the tuple as a property directly, and bypass the need to access the 'stats' property at all. Doing this will add a lot of properties to my schema, and unpack a few different sets of attributes which belong together logically. There are also other queries in which I'd like to affect the whole tuple at once, which would become somewhat painful under this approach.
2. Model the whole tuple as a new object type which can be linked to. This will eventually result in storing most of the Cartesian product of the domains of these fields as a new table, which also feels redundant. It does retain the logical packaging of these value groups though.

Is there a 'clean' EdgeQL way to perform these updates in place within the tuple? If not, which of the above alternatives would you recommend, or is there another approach to this modelling problem?

**Comment by raddevon** [comment, marked as answer]
You can always do this:

```
update Character
set {stats := (Strength := 1, Dexterity := .stats.Dexterity, Wisdom := .stats.Wisdom)}
```

I'm looking into whether there's a more elegant solution...

**Reply by realTPowell** [reply]
Thank you, this is probably the tidiest option while insisting on using the tuple. I wonder if it's possible to encapsulate that setter in a function at the top of the schema.

**Comment by raddevon** [comment]
For better ergonomics, you might consider modeling the stats as a separate type and linking to them from a character.

```
type Character { 
  required property name -> str;
  required link stats -> Stats;
}
type Stats {
  required property strength -> int16;
  required property dexterity -> int16
  required property wisdom -> int16;
}
```

If you can't do that for one reason or another, my previous answer is probably your best bet.

**Reply by realTPowell** [reply]
This is probably equally ergonomic in the end, my only question would be how costly the memory overhead of storing the links is
